In [75]:
# import libraries

import pandas as pd
import numpy as np

In [101]:
# open and observe file

file = pd.read_csv('../georgia_earthquakes.csv')
file.head()

time(utc)  magnitude(ml)  depth(km) lat/long(degree)  \
0  03/09/2023 16:35            5.2        107   40.3335/48.266   
1  26/08/2023 12:43            3.7          1    42.816/45.701   
2  24/08/2023 13:15            3.8          2     41.231/44.28   
3  14/08/2023 07:05            3.2         24  42.4708/45.0031   
4  28/07/2023 05:55            3.5          8  41.7878/43.1566   

                                              region  
0        Azerbaijan. From Georgia border - 162km.     
1   Russia. Chechnia. From Georgia border - 31km.     
2  From Tbilisi  54km. City Dmanisi - South-East ...  
3   Town Khazbegi - South-East - 33km. Village Ch...  
4   City Borjomi - West - 17km. Village Atskuri -...

In [102]:
# clean data by splitting columns and dropping unnececary ones

file[['Date', 'Time']] = file['time(utc)'].str.split(' ',expand = True)
file[['Latitude', 'Longitude']] = file['lat/long(degree)'].str.split('/', expand = True)
file = file.drop(['time(utc)', 'lat/long(degree)'], axis = 1)

In [103]:
# move colmns with Date and Time into the first positions of the DataFrame

first_col = file.pop('Date')
file.insert(0, 'Date', first_col)

second_col = file.pop('Time')
file.insert(1, 'Time', second_col)
file

Date   Time  magnitude(ml)  depth(km)  \
0    03/09/2023  16:35            5.2        107   
1    26/08/2023  12:43            3.7          1   
2    24/08/2023  13:15            3.8          2   
3    14/08/2023  07:05            3.2         24   
4    28/07/2023  05:55            3.5          8   
..          ...    ...            ...        ...   
281  25/07/2021  09:02            3.8         12   
282  20/07/2021  22:03            3.0         10   
283  16/07/2021  01:04            2.8         10   
284  14/07/2021  07:41            3.1         13   
285  14/07/2021  07:04            4.0         10   

                                                region Latitude Longitude  
0          Azerbaijan. From Georgia border - 162km.     40.3335    48.266  
1     Russia. Chechnia. From Georgia border - 31km.      42.816    45.701  
2    From Tbilisi  54km. City Dmanisi - South-East ...   41.231     44.28  
3     Town Khazbegi - South-East - 33km. Village Ch...  42.4708   45.0031  
4     City Borjomi - West - 17km. Village Atskuri -...  41.7878   43.1566  
..                                                 ...      ...       ...  
281  From Tbilisi  54km. City Tsalka - South-West -...  41.4611   44.0108  
282                                                NaN  43.4898    41.177  
283                                                     41.1725   43.9338  
284   City Ninotsminda - East - 25km. Village Sameb...  41.1673   43.8848  
285   City Ninotsminda - East - 21km. Village Sagha...  41.1793   43.9273  

[286 rows x 7 columns]

In [104]:
# rename columns for convenience

file.columns = map(str.capitalize,file.columns)

In [105]:
# search for duplicated rows
with_dups = file
print('Shape of table with duplicates: ', with_dups.shape)

without_dups = with_dups.drop_duplicates()
print('Shape of table without duplicates: ',without_dups.shape)

# we see, that there are no duplicated rows, so we can continue using original DataFrame 'file'

Shape of table with duplicates:  (286, 7)
Shape of table without duplicates:  (286, 7)


In [106]:
# check for data types in the DataFrame

print(file.dtypes, '\n\n')

# change Longitude and Latitude data types from object to float
file['Longitude'] = file['Longitude'].astype('float')
file['Latitude'] = file['Latitude'].astype('float')

print(file.dtypes)

Date              object
Time              object
Magnitude(ml)    float64
Depth(km)          int64
Region            object
Latitude          object
Longitude         object
dtype: object 


Date              object
Time              object
Magnitude(ml)    float64
Depth(km)          int64
Region            object
Latitude         float64
Longitude        float64
dtype: object


In [108]:
# check for missing data

print('Missing data: \n',file.isna().sum())

Missing data: 
 Date              0
Time              0
Magnitude(ml)     0
Depth(km)         0
Region           16
Latitude          0
Longitude         0
dtype: int64


In Region column 16 out of 286 region names are missed. We could 

In [120]:
print(file[file['Longitude']<30])
print(file[file['Latitude']<30])

Empty DataFrame
Columns: [Date, Time, Magnitude(ml), Depth(km), Region, Latitude, Longitude]
Index: []
Empty DataFrame
Columns: [Date, Time, Magnitude(ml), Depth(km), Region, Latitude, Longitude]
Index: []
